demos/minimal/actors/sample_processor.py

```diff
@@ -1,63 +1,77 @@
+        self.frame = None  # Initialize variable to store the current frame
+
+        context = zmq.Context()
+        self.socket = context.socket(zmq.PUB)
+        self.socket.bind("tcp://127.0.0.1:5555")
+
         logger.info("Completed setup for Processor")
 
     def stop(self):
-        """Trivial stop function for testing purposes."""
-
         logger.info("Processor stopping")
+        self.socket.close()
+        return 0
 
     def runStep(self):
-        """Gets from the input queue and calculates the average.
-
-        Receives an ObjectID, references data in the store using that
-        ObjectID, calculates the average of that data, and finally prints
-        to stdout.
         """
-
-        frame = None
+        Receives data ID from the queue, retrieves data from the Plasma store,
+        processes it, stores it in `processed_data`, and sends it through the
+        socket as flattened data with the frame number appended.
+        """
         try:
-            frame = self.q_in.get(timeout=0.001)
-
-        except:
-            logger.error("Could not get frame!")
-            pass
-
-        if frame is not None and self.frame_num is not None:
-            self.done = False
-            self.frame = self.client.getID(frame[0][0])
-            avg = np.mean(self.frame[0])
-
-            # print(f"Average: {avg}")
-            self.avg_list.append(avg)
-            # print(f"Overall Average: {np.mean(self.avg_list)}")
-            # print(f"Frame number: {self.frame_num}")
-            self.frame_num += 1
+            # Retrieve data ID from the queue
+            data_id = self.q_in.get(timeout=0.05)
+        except Empty:
+            return  # No data received, skip this step
+        except Exception as e:
+            logger.error(f"Error retrieving data ID: {e}")
+            return
+
+        if data_id is not None:
+            try:
+                # Fetch the data from the client using the ObjectID
+                self.frame = self.client.getID(data_id[0][0])  # Retrieve the frame data
+
+                # Unpack the flattened data
+                data = np.array(self.frame, dtype=np.float64)  # Ensure it's a NumPy array
+                values = data[:-1]  # Exclude the last element
+                frame_num = int(data[-1])  # Extract the last element as frame number
+
+                # Reshape values to 2D array (N, 2) for processing
+                values = values.reshape(-1, 2)
+
+                # Perform processing (e.g., scaling the y-values)
+                values[:, 1] *= 2  # Example: Scale y-coordinates by 2
+
+                # Flatten processed values and append frame number
+                self.processed_data = np.append(values.flatten(), frame_num)
+
+                # Send the processed data through the ZMQ socket
+                self.socket.send(self.processed_data.tobytes())
+                # logger.info(f"Frame {frame_num}: Sent {values.shape[0]} points after processing")
```

Comment:
I would add this back in to the logger, can be helpful for a user to see when they look at the logs that this step was reached if there is ever an error.